core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -127,6 +136,19 @@ public void setFileIOBuilder(Function<Map<String, String>, FileIO> newIOBuilder)
     this.ioBuilder = newIOBuilder;
   }
 
+  public void setClientPoolBuilder(
+      Function<Map<String, String>, JdbcClientPool> newClientPoolBuilder) {
+    Preconditions.checkState(
+        null == connections, "Cannot set client pool builder after calling initialize");
+    this.clientPoolBuilder = newClientPoolBuilder;
+  }
+
+  public void setInitializeCatalogTables(boolean initializeCatalogTables) {
```

Comment:
Why is this needed?

Reply:
Since we create the catalog class for every Trino query, we don’t want to run this repeatedly.

Reply:
Got it. We should just have a `initialized` flag on the constructor, too.